### PR TITLE
i18n(es): Update `api-reference.mdx`

### PR DESCRIPTION
**Pull request by Waxer59**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- What does this PR change? Give us a brief description.
Updated `api-reference.mdx`
* [9b16a32](https://github.com/withastro/docs/commit/9b16a329f95c0cfd2f3bae2e19876661cf66057a)
* [8dc000f](https://github.com/withastro/docs/commit/8dc000ff5c87fe9015890d15eda08c55e4a5a663)
* [b18535b](https://github.com/withastro/docs/commit/b18535bb3c541bad3a4f82fb8309869ec6fd6523)

### CODE REVIEW

File: src/content/docs/es/reference/api-reference.mdx [hunk 1]
```diff
@@ -746,6 +763,34 @@ El `schema` es un objeto de Zod opcional para configurar el tipo y la forma del
 
 [Consulta la guía de `Colección de Contenido`](/es/guides/content-collections/#defining-a-collection-schema) para ejemplos de uso.
 
+### `reference()`
+
+**Tipo:** `(collection: string) => ZodEffects<ZodString, { collection, id: string } | { collection, slug: string }>`
+
+La función `reference()` se utiliza en la configuración del contenido para definir una relación, o "referencia", de una colección a otra. Esta función acepta el nombre de la colección y valida el identificador/es de entrada especificados en el frontmatter o archivo de datos del contenido.
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:770]:
```suggestion
La función `reference()` se utiliza en la configuración del contenido para definir una relación, o "referencia" de una colección a otra. Esta función acepta el nombre de la colección y valida el identificador/es de entrada especificados en el frontmatter o archivo de datos del contenido.
```

File: src/content/docs/es/reference/api-reference.mdx [hunk 2]
```diff
@@ -746,6 +763,34 @@ El `schema` es un objeto de Zod opcional para configurar el tipo y la forma del
...
+
+Este ejemplo define referencias desde un autor de blog a la colección `authors` y a un array de publicaciones relacionadas en la misma colección `blog`:
+
+```ts
+import { defineCollection, reference, z } from 'astro:content';
+const blog = defineCollection({
+  type: 'content',
+  schema: z.object({
+    // Hacer referencia a un único autor de la colección `authors` mediante el campo `id`.
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:779]:
```suggestion
    // Hace referencia a un único autor de la colección `authors` mediante el campo `id`.
```

File: src/content/docs/es/reference/api-reference.mdx [hunk 3]
```diff
@@ -746,6 +763,34 @@ El `schema` es un objeto de Zod opcional para configurar el tipo y la forma del
...
+    author: reference('authors'),
+    // Hacer referencia a un array de publicaciones relacionadas de la colección `blog` mediante el campo `slug`.
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:781]:
```suggestion
    // Hace referencia a un arreglo de publicaciones relacionadas de la colección `blog` mediante el campo `slug`.
```

File: src/content/docs/es/reference/api-reference.mdx [hunk 4]
```diff
@@ -770,20 +815,76 @@ const draftBlogPosts = await getCollection('blog', ({ data }) => {
...
+- `({ collection: string, slug: string }) => CollectionEntry<collection>`
 
-`getEntry()` es una función que recupera una sola entrada de colección por nombre de colección e [identificador de entrada](#id).
+`getEntry()` es una función que recupera una única entrada de una colección mediante el nombre de la colección y ya sea el `id` de la entrada (para colecciones `type: 'data'`) o el `slug` de la entrada (para colecciones `type: 'content'`). `getEntry()` también se puede utilizar para obtener las entradas referenciadas y acceder a las propiedades `data`, `body` o `render()`:
 
 ```astro
 ---
 import { getEntry } from 'astro:content';
-const enterprise = await getEntry('blog', 'enterprise.md');
+// Obtener `src/content/blog/enterprise.md`
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:833]:
```suggestion
// Obtiene `src/content/blog/enterprise.md`
```

File: src/content/docs/es/reference/api-reference.mdx [hunk 5]
```diff
@@ -770,20 +815,76 @@ const draftBlogPosts = await getCollection('blog', ({ data }) => {
...
+const enterprisePost = await getEntry('blog', 'enterprise');
+// Obtener `src/content/captains/picard.yaml`
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:835]:
```suggestion
// Obtiene `src/content/captains/picard.yaml`
```

File: src/content/docs/es/reference/api-reference.mdx [hunk 6]
```diff
@@ -746,6 +763,34 @@ El `schema` es un objeto de Zod opcional para configurar el tipo y la forma del
 
 [Consulta la guía de `Colección de Contenido`](/es/guides/content-collections/#defining-a-collection-schema) para ejemplos de uso.
 
+### `reference()`
+
+**Tipo:** `(collection: string) => ZodEffects<ZodString, { collection, id: string } | { collection, slug: string }>`
+
+La función `reference()` se utiliza en la configuración del contenido para definir una relación, o "referencia", de una colección a otra. Esta función acepta el nombre de la colección y valida el identificador/es de entrada especificados en el frontmatter o archivo de datos del contenido.
+
+Este ejemplo define referencias desde un autor de blog a la colección `authors` y a un array de publicaciones relacionadas en la misma colección `blog`:
```
Comment (dreyfus92) [inline src/content/docs/es/reference/api-reference.mdx:772]:
```suggestion
Este ejemplo define referencias desde un autor de blog a la colección `authors` y a un arreglo de publicaciones relacionadas en la misma colección `blog`:
```

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
Buen trabajo @Waxer59, te dejo algunas sugerencias.

**Comment by Waxer59** [comment]
Gracias por la revisión 😃

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

**Pull request merged**